api/src/main/java/org/pmiops/workbench/api/OfflineUserController.java

```diff
@@ -201,4 +209,26 @@ private boolean timestampsEqual(Timestamp a, Timestamp b) {
     }
     return ResponseEntity.status(HttpStatus.NO_CONTENT).build();
   }
+
+  /**
+   * Audits GCP access for all users in the database.
+   *
+   * This API method is called by a cron job and is not part of our normal user-facing surface.
+   */
+  @Override
+  public ResponseEntity<Void> bulkAuditProjectAccess() {
+    int errorCount = 0;
+    int userCount = 0;
+
+    for (User user : userService.getAllUsers()) {
+      userCount++;
+      List<String> unauthorizedLogs = cloudResourceManagerService.getAllProjectsForUser(user).stream()
```

Comment:
Per discussion yesterday - this check is definitely much more limited in scope than is suggested in the ticket; for example if someone just adds me to a bucket, that project won't be returned by this API. Was there follow-up on this? Does this approach meet the acceptance criteria, or do we need to reconsider?

Reply:
Yep – see latest comments on https://precisionmedicineinitiative.atlassian.net/browse/RW-2062. This isn't perfect, but it's a first step. The security requirements were somewhat vague anyway – the main goal is to make a reasonable effort to detect unauthorized cross-org permissions. Project-level roles will at least capture the most pernicious abuses of these GSuite accounts.

Plus, it seems like there's a GCP API in development that will more directly hit our use case; rather than try to cobble something together, we may be better off just waiting for that.

@blrubenstein, maybe add a TODO(RW-2062) comment somewhere in this method to mark our intention to improve this if/when a better API becomes available?

Reply:
Will add the todo.